Source/FieldSolver/WarpXPushFieldsEM.cpp

```diff
@@ -372,82 +373,54 @@ WarpX::DampFieldsInGuards(std::array<std::unique_ptr<amrex::MultiFab>,3>& Efield
 
     for ( amrex::MFIter mfi(*Efield[0], amrex::TilingIfNotGPU()); mfi.isValid(); ++mfi )
     {
-
         amrex::Array4<amrex::Real> const& Ex_arr = Efield[0]->array(mfi);
         amrex::Array4<amrex::Real> const& Ey_arr = Efield[1]->array(mfi);
         amrex::Array4<amrex::Real> const& Ez_arr = Efield[2]->array(mfi);
         amrex::Array4<amrex::Real> const& Bx_arr = Bfield[0]->array(mfi);
         amrex::Array4<amrex::Real> const& By_arr = Bfield[1]->array(mfi);
         amrex::Array4<amrex::Real> const& Bz_arr = Bfield[2]->array(mfi);
 
-        // Get the tilebox from Efield so that it includes the guard cells.
-        amrex::Box tilebox = (*Efield[0])[mfi].box();
-        int const nz_tile = tilebox.bigEnd(zdir);
+        // Get the tileboxes from Efield and Bfield so that they include the guard cells.
+        amrex::Box tex = amrex::convert((*Efield[0])[mfi].box(), Efield[0]->ixType().toIntVect());
+        amrex::Box tey = amrex::convert((*Efield[1])[mfi].box(), Efield[1]->ixType().toIntVect());
+        amrex::Box tez = amrex::convert((*Efield[2])[mfi].box(), Efield[2]->ixType().toIntVect());
+        amrex::Box tbx = amrex::convert((*Bfield[0])[mfi].box(), Bfield[0]->ixType().toIntVect());
+        amrex::Box tby = amrex::convert((*Bfield[1])[mfi].box(), Bfield[1]->ixType().toIntVect());
+        amrex::Box tbz = amrex::convert((*Bfield[2])[mfi].box(), Bfield[2]->ixType().toIntVect());
 
         // Box for the whole simulation domain
         amrex::Box const& domain = Geom(0).Domain();
         int const nz_domain = domain.bigEnd(zdir);
 
-        if (tilebox.smallEnd(zdir) < 0) {
-
-            // Apply damping factor in guards cells below the lower end of the domain
-            int const nz_guard = -tilebox.smallEnd(zdir);
-
-            // Set so the box only covers the lower half of the guard cells
-            tilebox.setBig(zdir, -nz_guard/2-1);
-
-            amrex::ParallelFor(tilebox, Efield[0]->nComp(),
-            [=] AMREX_GPU_DEVICE (int i, int j, int k, int icomp)
+        amrex::ParallelFor(
+            tex, Efield[0]->nComp(), [=] AMREX_GPU_DEVICE (int i, int j, int k, int icomp)
```

Comment:
I didn't know that this can be done this way. I was expecting to see `ParallelFor(tex, tey, tez, `.

Reply:
I did not know this either.

I think this is the only interface available if we want to pass the range for the fourth component, together with the boxes. Which I think is needed in the RZ case, where there might be more than one element along the fourth axis.

I could not find any other possible interface of `amrex::ParallelFor` that accepts three boxes and three `ncomp` parameters, corresponding to each MultiFab. But I could be mistaken.

@WeiqunZhang Could you confirm this or comment on this?

Reply:
If it would help, the ncomp will be the same for all three arrays.